File: 2023_8/baekjoon/10988.py
import sys

str = sys.stdin.readline().strip()
length = len(str)
if length % 2 == 1:
    str = str[:length//2] + str[length//2 + 1:]
    length -= 1
print(1 if str[:length//2] == str[length - 1:length//2 - 1:-1] else 0)

File: 2023_8/baekjoon/10988.rs
use std::io::stdin;

fn main() {
    let mut buf = String::new();
    stdin().read_line(&mut buf).unwrap();
    let s: Vec<char> = buf.trim().chars().collect();

    let n = s.len();
    for i in 0..(n / 2) {
        if s[i] != s[n - i - 1] {
            println!("0");
            return;
        }
    }
    println!("1");
    return;
}
